File: rc-application/src/main/java/com/transkript/reportcard/business/service/StudentService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.StudentDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.Student;

import java.util.List;

public interface StudentService {
    EntityResponse addStudent(StudentDto studentDto);

    List<StudentDto> getStudents();

    StudentDto getStudent(Long id);

    EntityResponse deleteStudent(Long id);

    Student getStudentEntity(Long id);

    EntityResponse updateStudent(Long id, StudentDto studentDto);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/interf/StudentService.kt
package com.transkript.reportcard.business.service.interf

import com.transkript.reportcard.api.dto.StudentDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.Student

interface StudentService {
    fun addStudent(studentDto: StudentDto?): EntityResponse?
    val students: List<StudentDto?>?
    fun getStudent(id: Long?): StudentDto?
    fun deleteStudent(id: Long?): EntityResponse?
    fun getStudentEntity(id: Long?): Student?
    fun updateStudent(id: Long?, studentDto: StudentDto?): EntityResponse?
}
